src/world/area_kmr/kmr_12/events.c

```diff
@@ -0,0 +1,221 @@
+#include "kmr_12.h"
+
+GEN_EXIT_WALK_SCRIPT(exit_west, 60, 0, "kmr_07", 1);
+GEN_EXIT_WALK_SCRIPT(exit_east, 60, 1, "kmr_11", 0);
+
+Script bind_exits = {
+    SI_BIND(exit_west, TriggerFlag_FLOOR_ABOVE, 0 /* deili1 */, NULL),
+    SI_BIND(exit_east, TriggerFlag_FLOOR_ABOVE, 3 /* deili2 */, NULL),
+    SI_RETURN(),
+    SI_END(),
+};
+
+Script kmr_12_main = {
+    SI_SET(SI_SAVE_VAR(425), 31),
+    SI_CALL(SetSpriteShading, -1),
+    SI_CALL(SetCamPerspective, 0, 3, 25, 16, 4096),
+    SI_CALL(SetCamBGColor, 0, 0, 0, 0),
+    SI_CALL(SetCamEnabled, 0, 1),
+    SI_CALL(MakeNpcs, 0, npc_groups),
+    SI_EXEC_WAIT(make_entities),
+    SI_EXEC(kmr_12_play_music),
+    SI_SET(SI_VAR(0), bind_exits),
+    SI_EXEC(EnterWalk),
+    SI_WAIT_FRAMES(1),
+    SI_BIND(read_west_sign, TriggerFlag_WALL_INTERACT, 10, NULL),
+    SI_RETURN(),
+    SI_END(),
+};
+
+NpcAISettings goomba_ai_settings = {
+    .moveSpeed = 1.5f,
+    .moveTime = 30,
+    .waitTime = 30,
+    .alertRadius = 130.0f,
+    .unk_10 = 0.0f,
+    .unk_14 = 1,
+    .chaseSpeed = 2.5f,
+    .unk_1C = 180,
+    .unk_20 = 3,
+    .chaseRadius = 150.0f,
+    .unk_28 = 0.0f,
+    .unk_2C = TRUE,
+};
+
+Script goomba_ai = {
+    SI_CALL(DoBasicAI, &goomba_ai_settings),
+    SI_RETURN(),
+    SI_END(),
+};
+
+StaticNpcSettings goomba_npc_settings = {
+    .height = 20,
+    .radius = 23,
+    .aiScript = &goomba_ai,
+    .hitScript = (Bytecode*)0x80077F70,
+    .defeatScript = (Bytecode*)0x8007809C,
```

Comment:
Not a big fan of leaving raw pointers in C code.

Reply:
we probably should give these symbol names, even if they're like Script_8007whatever

Reply:
Yeah, I'm just not sure whether to name these `NormalEnemyHit` or something specific to Goombas. I have a feeling its the latter.